ada/views.py

```diff
@@ -0,0 +1,33 @@
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import HttpResponseRedirect, render
+
+from .models import RedirectUrl, RedirectUrlsEntry
+
+
+def clickActivity(request: HttpRequest, pk: int, whatsappid: int) -> HttpResponse:
+    try:
+        redirect_url = RedirectUrl.objects.get(id=pk)
+    except (ValueError, RedirectUrl.DoesNotExist):
+        return render(request, "index.html", {"error": "HTTP Bad Request"})
```

Comment:
We should probably just return a 404 in this case. There's a `get_object_or_404` method that we can use to easily do that: https://docs.djangoproject.com/en/3.2/topics/http/shortcuts/#get-object-or-404